x-pack/plugins/cases/server/saved_object_types/import_export/export.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  CoreSetup,
+  Logger,
+  SavedObject,
+  SavedObjectsClientContract,
+  SavedObjectsExportTransformContext,
+} from 'kibana/server';
+import {
+  CaseUserActionAttributes,
+  CASE_COMMENT_SAVED_OBJECT,
+  CASE_SAVED_OBJECT,
+  CASE_USER_ACTION_SAVED_OBJECT,
+  CommentAttributes,
+  MAX_DOCS_PER_PAGE,
+  SAVED_OBJECT_TYPES,
+} from '../../../common';
+import { createCaseError, defaultSortField } from '../../common';
+import { ESCaseAttributes } from '../../services/cases/types';
+
+export async function handleExport({
+  context,
+  objects,
+  coreSetup,
+  logger,
+}: {
+  context: SavedObjectsExportTransformContext;
+  objects: Array<SavedObject<ESCaseAttributes>>;
+  coreSetup: CoreSetup;
+  logger: Logger;
+}): Promise<Array<SavedObject<ESCaseAttributes | CommentAttributes | CaseUserActionAttributes>>> {
+  try {
+    if (objects.length <= 0) {
+      return [];
+    }
+
+    const [{ savedObjects }] = await coreSetup.getStartServices();
+    const savedObjectsClient = savedObjects.getScopedClient(context.request, {
```

Comment:
I'm creating the saved objects client directly here instead of using our service layer because the service layer can mutate the data (for example creating the connector_id field when the objects are returned). We can also make more performant calls here for both the attachments and user actions at the same time using point in time searches instead of the implementation that the service layer uses.